common/optimiser/Optimiser.java

```diff
@@ -57,12 +57,8 @@ public Optimiser() {
     }
 
     public synchronized Status optimise(long timeLimitMillis) {
-        if (hasSolver && constraintsChanged()) {
-            constraints.forEach(OptimiserConstraint::updateCoefficients);
-            status = Status.NOT_SOLVED;
-        }
+        assert hasSolver;
         if (isOptimal()) return status;
-        else if (!hasSolver) initialiseSolver();
```

Comment:
`optimise()` no longer needs to initialise the solver, `GraphPlanner` takes care of that when it sets or updates the model.